### PR TITLE
components spring17 cleaning post

### PR DESCRIPTION
**Pull request by stdweird**

### CODE REVIEW

File: _posts/2017-02-10-components-spring17-cleaning.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+# 00-tqu
+
+We introduce a new basic perl unittest called `00-tqu.t` referring to
+the usage of the [`Test::Quattor::Unittest` tqu] module.
+
+Adding it is easy:
+    echo 'use Test::Quattor::Unittest;' > src/test/perl/00-tqu.t
+
+The unittests replaces a number of (posisble) existsing tests that can be removed
```
Comment (jrha) [inline _posts/2017-02-10-components-spring17-cleaning.md]: unit tests
replace
possible
existing

File: _posts/2017-02-10-components-spring17-cleaning.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+   * generates code with a prefix
+   * includes the schema bind
+
+3. `schema.pan`
+ * replace header (incl declaration) with `${componentschema}`
+ * try to get rid of `include 'quattor/schema';`
+  * components typically only requires `include 'quattor/types/component';`
+ * remove the bind (it is in the `config.pan` now)
+ * rename the main component type to `${project.artifactId}_component` (this is the type used in `config.pan`)
+ * check if there is a unittest that tests config.pan
```
Comment (jrha) [inline _posts/2017-02-10-components-spring17-cleaning.md]: unit test

File: _posts/2017-02-10-components-spring17-cleaning.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+  * a perl unittest that has `use Test::Quattor qw(<unittestobjecttemplate>);`
```
Comment (jrha) [inline _posts/2017-02-10-components-spring17-cleaning.md]: unit test

File: _posts/2017-02-10-components-spring17-cleaning.md [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+layout: article
+title: Components Spring 2017 Cleaning
+author: Stijn De Weirdt
+category: documentation
+---
+
+# Goal
+
+With the advent of build-tools version 1.51,
+some code cleanup and refactoring is recommended.
```
Comment (jrha) [inline _posts/2017-02-10-components-spring17-cleaning.md]: clean up

File: _posts/2017-02-10-components-spring17-cleaning.md [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+
+This document explains how to cleanup the components;
```
Comment (jrha) [inline _posts/2017-02-10-components-spring17-cleaning.md]: clean up

File: _posts/2017-02-10-components-spring17-cleaning.md [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+the cleanup of the other core modules is not handled.
+
+The minimal requirement is to bump `build-tools` in `pom.xml`
+to at least 1.51 (under `/project/parent/version`).
```
Comment (jrha) [inline _posts/2017-02-10-components-spring17-cleaning.md]: The easiest way to do this is run `mvn versions:update-parent`

File: _posts/2017-02-10-components-spring17-cleaning.md [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+
+# Pan cleanup
+
+1. no more `-common` and / or `-rpms` ; most components have only config and schema
+2. `config.pan`
+  * use config headers `${componentconfig}`
+   * has spma as default predependency and last entry
+    * use template `${componentconfignospma}` to set different dependencies
+   * generates code with a prefix
+   * includes the schema bind
```
Comment (ned21) [inline _posts/2017-02-10-components-spring17-cleaning.md]: I think we agreed at the last workshop to also start including:
```
include { if_exists('site-config') };
```
to allows sites to provide a per-component custom config?

File: _posts/2017-02-10-components-spring17-cleaning.md [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+
+It also adds a `critic` test and a `perltidy` test. (For now, the `perltidy` test always passes).
+
+[tqu]: http://quattor-documentation.readthedocs.io/en/latest/Unittesting/Quattor::Unittest/
+
+# Perl module cleanup
+
+1. replace header with PMpre/post header (space after `PMpre` is required)
+
+    #${PMpre} NCM::Component::${project.artifactId}${PMpost}
```
Comment (ned21) [inline _posts/2017-02-10-components-spring17-cleaning.md]: The GH previewer is rendering this badly. Is that just a problem with GH and looks OK when using whatever the website uses?

File: _posts/2017-02-10-components-spring17-cleaning.md (hunk 7 above)
Comment (stdweird) [inline _posts/2017-02-10-components-spring17-cleaning.md]: it's there: https://github.com/quattor/configuration-modules-core/pull/1031#issuecomment-278923962. i'll add a url to the releated sections in the build-profile pom.xml

**Comment by stdweird** [comment]
reworked the text a bit so renders nicely
@jrha this now assumes build-tools 1.52

**Comment by jrha** [comment]
1.52 has been built and should show up on Nexus soonish.

**Comment by stdweird** [comment]
1.52 is out, first PR is coming up 😄

**Comment by stdweird** [comment]
@jrha travis fails on 2 broken links from other articles....

**Pull request merged**